ui/src/app/machines/views/MachineList/MachineListHeader/ActionFormWrapper/DeployForm/DeployFormFields/DeployFormFields.test.tsx

```diff
@@ -259,17 +239,30 @@ describe("DeployFormFields", () => {
         <MemoryRouter
           initialEntries={[{ pathname: "/machines/add", key: "testKey" }]}
         >
-          <DeployForm
-            processing={false}
-            setProcessing={jest.fn()}
-            setSelectedAction={jest.fn()}
-          />
+          <DeployForm setSelectedAction={jest.fn()} />
         </MemoryRouter>
       </Provider>
     );
     expect(wrapper.find("[data-test='sshkeys-warning']").exists()).toBe(true);
   });
 
+  it("displays an error if there are no OSes or releases to choose from", () => {
+    const state = { ...initialState };
+    state.general.osInfo.data.osystems = [];
+    state.general.osInfo.data.releases = [];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines/add", key: "testKey" }]}
+        >
+          <DeployForm setSelectedAction={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("[data-test='images-error']").exists()).toBe(true);
```

Comment:
Might also be good to check the fields are disabled.

Reply:
Done